h2/src/test/org/h2/test/unit/TestClearReferences.java

```diff
@@ -30,6 +30,9 @@
         "org.h2.expression.function.DateTimeFunctions.WEEK_FIELDS",
         "org.h2.expression.function.ToChar.NAMES",
         "org.h2.jdbcx.JdbcDataSourceFactory.cachedTraceSystem",
+        "org.h2.mvstore.db.ValueDataType.compareMode",
+        "org.h2.mvstore.type.LongDataType.DUMMY",
+        "org.h2.mvstore.type.LongDataType.INSTANCE",
```

Comment:
They aren't refreshed automatically and cannot be listed here.

Reply:
I don't understand why instance field `ValueDataType.compareMode` is cleared, that's very weird.

Reply:
Both fields from `LongDataType` aren't detected by this test and can be simply removed from the list. But I don't know what to do with `compareMode`.

Reply:
Honestly, I have no idea what is is all about. What this "refreshed automatically" means?  Why are we dealing here with some Tomcat idiosyncrasies (which BTW might be a thing of a past)? What JVM spec is broken by having such static field?

Reply:
Older versions of Tomcat had some idiosyncracies when doing hot reload of web applications, which this stuff works around, not sure if it is still necessary

Reply:
It means that code reads the field and if it was set to `null` it sets in again to a value that it should have, for example, a lost cache can be created again, it will be empty, but it's not a disaster.

If value can't be recovered, another approach is needed, such as use `Object` type of the field and typecasts, `Object` fields aren't cleared.

But `ValueDataType.compareMode` isn't a static field and I don't understand why it is detected as cleared by Tomcat.

Reply:
> Why are we dealing here with some Tomcat idiosyncrasies

See cde29719934e2ea98855251732368313140e42b9.

I don't like that workaround too.

> which BTW might be a thing of a past

This code still exists in Tomcat. I'm not sure is it enabled by default or not.

Reply:
Actually recent versions of Tomcat have a redesigned cleanup code with multiple configuration options. `org.apache.catalina.loader. WebappClassLoader.ENABLE_CLEAR_REFERENCES` was removed.

Reply:
I removed `TestClearReferences` in #2296.